### PR TITLE
feat!: default for `enforceForClassMembers` in `no-useless-computed-key`

### LINKED ISSUE #18042: Rule Change: Change default for `no-useless-computed-key`'s `enforceForClassMembers` option to `true`
### What rule do you want to change?

[no-useless-computed-key](https://eslint.org/docs/latest/rules/no-useless-computed-key)

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint no-useless-computed-key: "error"*/

class Foo {
    ["foo"] = "bar";

    [0]() {}
    ['a']() {}
    get ['b']() {}
    set ['c'](value) {}

    static ["foo"] = "bar";

    static ['a']() {}
}
```

### What does the rule currently do for this code?

No problem is reported, because the `enforceForClassMembers` defaults to `false`.

### What will the rule do after it's changed?

Report several problems like in [this rule example](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVzZWxlc3MtY29tcHV0ZWQta2V5OiBbXCJlcnJvclwiLCB7IFwiZW5mb3JjZUZvckNsYXNzTWVtYmVyc1wiOiB0cnVlIH1dKi9cblxuY2xhc3MgRm9vIHtcbiAgICBbXCJmb29cIl0gPSBcImJhclwiO1xuXG4gICAgWzBdKCkge31cbiAgICBbJ2EnXSgpIHt9XG4gICAgZ2V0IFsnYiddKCkge31cbiAgICBzZXQgWydjJ10odmFsdWUpIHt9XG5cbiAgICBzdGF0aWMgW1wiZm9vXCJdID0gXCJiYXJcIjtcblxuICAgIHN0YXRpYyBbJ2EnXSgpIHt9XG59In0=), but without the option explicitly set.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

This change has been already suggested when the `enforceForClassMembers` option was introduced: https://github.com/eslint/eslint/issues/12041#issuecomment-516802519. Like in other cases, the very reason for having this option was to not introduce a breaking change at that time. This change could be done before v9 is released.

### PR DESCRIPTION
**Pull request by fasttime**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Fixes #18042

#### What changes did you make? (Give an overview)

This PR changes the default value of the `enforceForClassMembers` option in the `no-useless-computed-key` rule to `true`.

Since this is a breaking change in v9, I've updated the migration guide.

### CODE REVIEW

**Comment by Tanujkanti4441** [comment]
Can you also add `To address` section in the migration guide to tell the user how previous behavior can be achieved?

**Comment by Tanujkanti4441** [comment]
issue #12048 is also about `no-useless-computed-key` rule, how about cover it too in this PR, if @mdjermanovic confirms that the behavior of the rule is correct.

**Comment by mdjermanovic** [comment]
> issue #12048 is also about `no-useless-computed-key` rule, how about cover it too in this PR, if @mdjermanovic confirms that the behavior of the rule is correct.

I think it should be a separate PR as it isn't related to this option.

File: docs/src/use/migrate-to-9.0.0.md [hunk 1]
```diff
@@ -362,6 +363,21 @@ if (test) {
 
 **Related issue(s):** [#15576](https://github.com/eslint/eslint/issues/15576)
 
+## <a name="no-useless-computed-key"></a> `no-useless-computed-key` flags unnecessary computed member names in classes by default
+
+In ESLint v9.0.0, the default value of the `enforceForClassMembers` option of the `no-useless-computed-key` rule was changed from `false` to `true`.
+The effect of this change ist that unnecessary computed member names in classes will be flagged by default.
```
Comment (mdjermanovic) [inline docs/src/use/migrate-to-9.0.0.md:369]:
```suggestion
The effect of this change is that unnecessary computed member names in classes will be flagged by default.
```

**Comment by nzakas** [comment]
Folks: please remember to move PRs through the triage board:
https://eslint.org/docs/latest/maintain/manage-issues#when-an-issue-or-pull-request-is-opened

**Comment by mdjermanovic** [comment]
There are merge conflicts in the migration guide now.

**Review by mdjermanovic — APPROVED** [review]
LGTM when merge conflicts are fixed.

**Comment by fasttime** [comment]
The merge conflicts are fixed.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**